guillotina/contrib/pubsub/utility.py

```diff
@@ -76,10 +75,14 @@ async def real_subscribe(self, channel_name):
                 return
             except Exception:
                 logger.error(f"Unhandled exception with pubsub. Sleeping before trying again", exc_info=True)
+                # TODO: maybe we should call the callback with a disconnected event or do it on reconnect,
+                # so the callback has a chance to perform the logic to recover.
```

Comment:
@bloodbare @vangheem what do you think about creating a new way to register callbacks like:

`utility.subscribe(chan_name, rid, callback, all_events=False)`

When the callback is registered with `all_events=True`, the callback signature will be `callback(event_name, **msg)`, so we can notify the events `on_message`, `on_disconnect`, ... and the callback can perform the desired logic accordingly.

In particular, I want to fix the cache. We should clear the entire cache when the channel is disconnected/connected, because we'll never receive the messages published while it was disconnected, therefore, losing invalidations and having stale data.